EmployeeDirectory.BAL/Providers/EmployeeProvider.cs

```diff
@@ -140,15 +165,16 @@ public List<Employee> GetEmployees()
                 if (employeeIndex != -1)
                 {
                     employeeId = employeeList[employeeIndex].Id;
+                    deptId = employeeList[employeeIndex].Department;
                     return (true, employeeList[employeeIndex]);
                 }
                 throw new EmpNotFound("Employee Not found with given Id");
             }
-            catch (IOException)
+            catch (EmpNotFound)
```

Comment:
Don't use try catch blocks in BAL or DAL. Use it only in Base Project (i.e., Console Project). Although you can throw some exceptions from child projects but need to catch only in Console project. (Make this change to complete application).